src/js/components/SelectMultiple/SelectionSummary.js

```diff
@@ -54,28 +56,30 @@ const SelectionSummary = ({
       >
         <Box pad={{ vertical: 'xsmall' }} alignSelf="center">
           <Text margin={{ vertical: 'xsmall' }} size="small">
-            {value.length === 0 || onMore
-              ? `${value.length} selected`
-              : `${value.length} selected of ${options.length}`}
+            {value?.length === 0 || onMore || !value
+              ? `${value?.length || '0'} selected`
+              : `${value?.length || '0'} selected of ${options.length}`}
           </Text>
         </Box>
         {(options.length &&
-          (!limit || !(value.length === 0 && selectedValuesDisabled()))) > 0 &&
-          (!onMore || (onMore && value.length !== 0)) && (
+          (!limit ||
+            !(!value || (value?.length === 0 && selectedValuesDisabled())))) >
```

Comment:
Lines 66, 71, 76, 82 all have the same condition and we are repeating ourselves. Can we assign the result of the condition to a variable and use the var instead?

Reply:
Good suggestion
Added in latest commit